### PR TITLE
New style trigo

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/math/detail/generic/rempio2_kernel.hpp [hunk 1]
```diff
@@ -35,14 +41,15 @@ EVE_FORCEINLINE kumi::tuple<T, T, T>
     auto                a   = y - da;
     da                      = (y - a) - da;
 
-    return {n, a, da};
+    return kumi::make_tuple(n, a, da);
   }
   else if constexpr( std::is_same_v<elt_t, float> )
   {
+//    using i_t =  as_integer_t<T>;
```
Comment (jfalcou) [inline include/eve/module/math/detail/generic/rempio2_kernel.hpp:48]: remove don't comment

File: include/eve/module/math/regular/sech.hpp [hunk 2]
```diff
@@ -7,10 +7,25 @@
...
+#include <eve/module/math/regular/exp.hpp>
+#include <eve/module/math/constant/log_2.hpp>
+#include <eve/module/math/constant/maxlog.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct sech_t : elementwise_callable<sech_t, Options>
+  {
+    template<eve::value T>
```
Comment (jfalcou) [inline include/eve/module/math/regular/sech.hpp:23]: `floating_value` ?

File: include/eve/module/math/regular/sinh.hpp [hunk 3]
```diff
@@ -7,10 +7,25 @@
...
+#include <eve/module/math/constant/maxlog.hpp>
+#include <eve/module/math/regular/exp.hpp>
+#include <eve/module/math/regular/expm1.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct sinh_t : elementwise_callable<sinh_t, Options>
+  {
+    template<eve::value T>
```
Comment (jfalcou) [inline include/eve/module/math/regular/sinh.hpp:23]: `floating_value`

File: include/eve/module/math/regular/sinhc.hpp [hunk 4]
```diff
@@ -7,10 +7,24 @@
...
+#include <eve/module/core.hpp>
+#include <eve/module/math/regular/sinh.hpp>
+#include <eve/module/math/regular/exp.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct sinhc_t : elementwise_callable<sinhc_t, Options>
+  {
+    template<eve::value T>
```
Comment (jfalcou) [inline include/eve/module/math/regular/sinhc.hpp:22]: `floating_value`

File: include/eve/module/math/regular/tand.hpp [hunk 5]
```diff
@@ -7,10 +7,28 @@
...
+#include <eve/module/math/regular/div_180.hpp>
+#include <eve/module/math/regular/tanpi.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct tand_t : elementwise_callable<tand_t, Options, quarter_circle_option, half_circle_option,
+                                        full_circle_option, medium_option, big_option>
+  {
+    template<eve::value T>
```
Comment (jfalcou) [inline include/eve/module/math/regular/tand.hpp:26]: `floating_value`

File: include/eve/module/math/regular/tanh.hpp [hunk 6]
```diff
@@ -7,10 +7,23 @@
...
+#include <eve/module/core/decorator/core.hpp>
+#include <eve/module/core.hpp>
+#include <eve/module/math/regular/expm1.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct tanh_t : elementwise_callable<tanh_t, Options>
+  {
+    template<eve::value T>
```
Comment (jfalcou) [inline include/eve/module/math/regular/tanh.hpp:21]: `floating_value`

File: include/eve/module/math/regular/tanpi.hpp [hunk 7]
```diff
@@ -7,10 +7,27 @@
...
+#include <eve/module/math/regular/cos.hpp>
+#include <eve/module/math/detail/generic/rem2.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct tanpi_t : elementwise_callable<tanpi_t, Options, quarter_circle_option, half_circle_option,
+                                        full_circle_option, medium_option, big_option>
+  {
+    template<eve::value T>
```
Comment (jfalcou) [inline include/eve/module/math/regular/tanpi.hpp:25]: `floating_value`

**Review by jfalcou — CHANGES_REQUESTED** [review]

**Comment by jfalcou** [comment]
cospi shoudl not take value either to be symmetric

File: include/eve/module/math/detail/generic/rempio2_kernel.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/math/detail/generic/rempio2_kernel.hpp:48]: done

File: include/eve/module/math/regular/sech.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/math/regular/sech.hpp:23]: done

File: include/eve/module/math/regular/sinh.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/math/regular/sinh.hpp:23]: done

File: include/eve/module/math/regular/sinhc.hpp (hunk 4 above)
Comment (jtlap) [reply include/eve/module/math/regular/sinhc.hpp:22]: done

File: include/eve/module/math/regular/tand.hpp (hunk 5 above)
Comment (jtlap) [reply include/eve/module/math/regular/tand.hpp:26]: done

File: include/eve/module/math/regular/tanh.hpp (hunk 6 above)
Comment (jtlap) [reply include/eve/module/math/regular/tanh.hpp:21]: done

File: include/eve/module/math/regular/tanpi.hpp (hunk 7 above)
Comment (jtlap) [reply include/eve/module/math/regular/tanpi.hpp:25]: done

**Pull request merged**